src/ZipFile.jl

```diff
@@ -390,9 +397,10 @@ function close(w::Writer)
     _writele(w._io, UInt32(cdpos))
     _writele(w._io, UInt16(0))
 
-    if w._close_io
-        close(w._io)
-    end
+    flush(w._io)
+    seek(w._io, cdpos)
```

Comment:
Some explanation would be helpful for future readers. Please add a comment like this:
```
# Seek to the beginning of central directory so that appending more files will overwrite it.
```